IS0401Test.py

```diff
@@ -681,6 +664,65 @@ def test_19(self):
 
         return test.PASS()
 
+    def test_20(self):
+        """Node's resources correctly signal the current protocol"""
+
+        test = Test("Node's resources correctly signal the current protocol")
+
+        service_href_warn = False
+        device_href_warn = False
+
+        api = self.apis[NODE_API_KEY]
+        valid, response = self.do_request("GET", self.node_url + "self")
+        if not valid:
+            return test.FAIL("Unexpected response from the Node API: {}".format(response))
+        try:
+            node_self = response.json()
+            if not node_self["href"].startswith(self.protocol + "://"):
+                return test.FAIL("Node 'href' does not match the current protocol")
+            if self.is04_utils.compare_api_version(api["version"], "v1.1") >= 0:
+                for endpoint in node_self["api"]["endpoints"]:
+                    if endpoint["protocol"] != self.protocol:
+                        return test.FAIL("One or more Node 'endpoints' do not match the current protocol")
+            for service in node_self["services"]:
+                if not service["href"].startswith(self.protocol):
+                    # Only warn about these at the end so that more major failures are flagged first
+                    service_href_warn = True
+        except json.decoder.JSONDecodeError:
+            return test.FAIL("Non-JSON response returned from Node API")
+
+        if self.is04_utils.compare_api_version(api["version"], "v1.1") >= 0:
+            valid, response = self.do_request("GET", self.node_url + "devices")
+            if not valid:
+                return test.FAIL("Unexpected response from the Node API: {}".format(response))
+            try:
+                node_devices = response.json()
+                for device in node_devices:
+                    for control in device["controls"]:
+                        if not control["href"].startswith(self.protocol):
+                            # Only warn about these at the end so that more major failures are flagged first
+                            device_href_warn = True
+            except json.decoder.JSONDecodeError:
+                return test.FAIL("Non-JSON response returned from Node API")
+
+        valid, response = self.do_request("GET", self.node_url + "senders")
+        if not valid:
+            return test.FAIL("Unexpected response from the Node API: {}".format(response))
+        try:
+            node_senders = response.json()
+            for sender in node_senders:
+                if sender["manifest_href"] != "" and not sender["manifest_href"].startswith(self.protocol):
+                    return test.FAIL("One or more Sender 'manifest_href' values do not match the current protocol")

```

Comment:
I completely agree with Node service 'href' being a warning. And I think it makes sense for the Device control 'href' too - unless we want to mandate e.g. that Connection API protocol should match Node API protocol?

Should the 'manifest_href' be a warning too? I'm not sure, but it seems similar to the Connection API control case?

Reply:
I'll use a warning for now and we can re-visit if necessary. I think the Node/Connection API probably should match in protocol, but that's probably a check which should go into IS0502.